### PR TITLE
EA PG CronHistory - Part 1

### PR DESCRIPTION
**Pull request by oetherington**
This migrates the first "secret" collection to Postgres which is managed by the vendored synced cron library. The simplest way to do this seemed to just be to create a collection.

Part 2 is in #6523 

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1203783402136060) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/server/vendor/synced-cron/synced-cron-server.ts [hunk 1]
```diff
@@ -90,8 +90,7 @@ onStartup(function() {
     later.date.localTime();
 
   // collection holding the job history records
-  SyncedCron._collection = new MongoCollection(options.collectionName);
-  SyncedCron._collection._ensureIndex({intendedAt: 1, name: 1}, {unique: true});
+  SyncedCron._collection = CronHistories;
```
Comment (jpaddison3) [inline packages/lesswrong/server/vendor/synced-cron/synced-cron-server.ts:93]: Amazing that that's all it took.

File: packages/lesswrong/server/scripts/migrateCollections.ts [hunk 2]
```diff
@@ -111,10 +111,14 @@ const pickBatchSize = (collection: SwitchingCollection<DbObject>) => {
...
-  createdSince
-    ? { createdAt: { $gte: createdSince } }
-    : {};
+const makeBatchFilter = (collectionName: string, createdSince?: Date) => {
+  if (!createdSince) {
+    return {};
+  }
+  return collectionName === "cronHistory"
+    ? { startedAt: { $gte: createdSince } }
+    : { createdAt: { $gte: createdSince } };
```
Comment (jpaddison3) [inline packages/lesswrong/server/scripts/migrateCollections.ts:120]: yak-shaving.jpg

**Review by jpaddison3 — COMMENTED** [review]
Absolutely amazing. I mean, test and then let's f*****g ship it.
Please test:
 * That it's running cron jobs on your local machine
 * You're seeing logs saying "not running duplicate whatever" messages when running two local dev instances
 * On both mongo and postgres please

**Review by jpaddison3 — APPROVED** [review]

**Comment by oetherington** [comment]
Great!

Testing and working in Postgres:
Screenshot 2023-01-30 at 14 43 34
and Mongo:
Screenshot 2023-01-30 at 14 46 51

**Comment by jpaddison3** [comment]
@oetherington, I'm getting giant errors when ""not running updateScore"":
```
2023-01-30T16:03:00.655Z: SQL Error for CronHistories: duplicate key value violates unique constraint "idx_CronHistories_intendedAt_name": `INSERT INTO "CronHistories" ( "_id" , "intendedAt" , "name" , "startedAt" , "finishedAt" , "result" ) VALUES  ( $1 ,  $2 ,  $3 ,  $4 ,  $5 ,  $6 ) RETURNING *`: [
  'xNCuyKY6gq2fphLwq',
  2023-01-30T16:03:00.000Z,
  'updateScoreActiveDocuments',
  2023-01-30T16:03:00.621Z,
  null,
  null
]: {
  collectionName: 'CronHistories',
  data: {
    intendedAt: 2023-01-30T16:03:00.000Z,
    name: 'updateScoreActiveDocuments',
    startedAt: 2023-01-30T16:03:00.621Z,
    _id: 'xNCuyKY6gq2fphLwq'
  },
  options: undefined
}
```
Presumably everything's fine, but that's a lot of log spam.

**Comment by oetherington** [comment]
Ah, yeah. Everything _is_ fine - it's configured to report the error as soon as it gets back from Postgres before we have a chance to `catch` it, but I agree that's too much spam. I'll add it to the todo list.

**Comment by oetherington** [comment]
Log spam will be fixed by #6605

**Pull request merged**